auto_dev/mech/create_mech_tool.py

```diff
@@ -0,0 +1,320 @@
+    if it does not already exist.
+    Args:
+        tool_path (str): The path where the __init__.py file will be created.
+    Returns:
+        None
+    """
+    init_file_path = os.path.join(tool_path, '__init__.py')
+    if os.path.exists(init_file_path):
+        print(f"__init__.py already exists at {init_file_path}. Skipping creation.")
+        return
+
+    init_content = '''#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2024 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+'''
+    with open(init_file_path, 'w') as f:
+        f.write(init_content)
+    print(f"__init__.py created at {init_file_path}")
+
+
+def create_customs_folder(tool_path):
+    """
+    Creates a 'customs' folder within the given tool directory.
+    Args:
+        tool_path (str): The path to the tool directory.
+    Returns:
+        str: The path to the created 'customs' folder.
+    """
+    customs_path = os.path.join(tool_path, 'customs')
+    if not os.path.exists(customs_path):
+        os.makedirs(customs_path, exist_ok=True)
+        print(f"'customs' folder created at {customs_path}")
+    else:
+        print(f"'customs' folder already exists at {customs_path}")
+    return customs_path
+
+def create_tool_folder(customs_path, tool_name):
+    """
+    Creates a folder inside the customs folder with the name of the tool.
+    Args:
+        customs_path (str): The path to the customs folder.
+        tool_name (str): The name of the tool.
+    Returns:
+        str: The path to the created tool folder.
+    """
+    tool_folder_path = os.path.join(customs_path, tool_name)
+    if not os.path.exists(tool_folder_path):
+        os.makedirs(tool_folder_path, exist_ok=True)
+        print(f"Tool folder '{tool_name}' created at {tool_folder_path}")
+    else:
+        print(f"Tool folder '{tool_name}' already exists at {tool_folder_path}")
+    return tool_folder_path
+
+def create_component_yaml(tools_folder_path, tool_name, author_name):
+    yaml_path = os.path.join(tools_folder_path, 'component.yaml')
+    if os.path.exists(yaml_path):
+        print(f"component.yaml already exists at {yaml_path}. Skipping creation.")
+        return
+
+    component_yaml_content = f'''name: {tool_name}
```

Comment:
lets use jinja such that can have a folder that gets imputed with the collected vars